### PR TITLE
Add GitHub Actions for comprehensive testing and modernise packaging

### PR DESCRIPTION
**Pull request by brocksam**
This PR improves the use of GitHub Actions by Titlecase and modernises the packaging of Titlecase.

Regarding GitHub Actions, this PR adds (or improves) the workflows:
- `test.py`: try installing the package and run the tests on Ubuntu, macOS and Windows for the latest Python version (run on all pushes);
- `ci.py`: try installing the package and run the tests on Ubuntu, macOS and Windows for all supported Python version (run only on pull requests);
- `coverage.yml`: run the tests and update the coverage with Coveralls; and
- `publish.yml`: automatically publish the package to PyPI when a new GitHub release is made (note that this only works correctly with versioning when the additional package modernisation changes are also made).

Regarding the package modernisation, this PR takes packaging recommendations inline with [PEP 517](https://www.python.org/dev/peps/pep-0517/)/[PEP 518](https://www.python.org/dev/peps/pep-0518/), including:
- using `setup.cfg` for package metadata;
- adding a `pyproject.toml` file;
- shrinking `setup.py`;
- removing `MANIFEST.in`; and
- use [setuptools-scm](https://pypi.org/project/setuptools-scm/) for controlling the package versioning (which basically relies on the GitHub release tags - required in conjunction with `publish.yml`).

This PR also:
- changes the license file type to `.txt` as it is plain text; and
- removes the `.travis.yml` file as TravisCI seems to no longer work for this project.

Note to @ppannuto: for `publish.yml` to work, it is required that an [API token for Titlecase on PyPI is created](https://github.com/pypa/gh-action-pypi-publish) and [added to this repository as a secret called "PYPI_TOKEN"](https://docs.github.com/en/actions/reference/encrypted-secrets).

EDIT: an API token for Titlecase on Coveralls is also required. This must be added as a secret called "COVERALLS_TOKEN".

### CODE REVIEW

**Comment by brocksam** [comment]
I have also removed Python 3.4 and 3.5 as their end of life dates were 18th March 2019 and 13th September 2020 respectively.

**Comment by brocksam** [comment]
Waiting for further input from @fireundubh on issue #86 to try and resolve the test failures on Windows with Python 3.6 and 3.7 before this is taken further.

**Comment by brocksam** [comment]
This is ready to be merged (pending creation of the PyPI and Coveralls tokens as explained above) as I believe it will greatly help with the resolution of issue #86.

**Comment by brocksam** [comment]
Recommendation from @fireundubh on issue #86 implemented in https://github.com/ppannuto/python-titlecase/pull/85/commits/e7ca9fbbb519c8e2fafa6dcb3f0d7885b5ca313a. All tests now passing on Windows for all supported Python versions.

I suggest rerunning the `coverage.yml` workflow once the suggested tokens have been created and added to this repo as secrets before merging the PR to ensure that it's working correctly. Assuming that that is successful, this PR is now good to go.

File: titlecase/tests.py [hunk 1]
```diff
@@ -372,15 +372,17 @@ def test_set_small_word_list():
...
-    with tempfile.NamedTemporaryFile(mode='w') as f:
-        f.write('UDP\nPPPoE\n')
-        f.flush()
-        # This works without a wordlist, because it begins mixed case
-        assert titlecase('sending UDP packets over PPPoE works great') == 'Sending UDP Packets Over PPPoE Works Great'
-        # Without a wordlist, this will do the "wrong" thing for the context
-        assert titlecase('SENDING UDP PACKETS OVER PPPOE WORKS GREAT') == 'Sending Udp Packets Over Pppoe Works Great'
-        # A wordlist can provide custom acronyms
-        assert titlecase('sending UDP packets over PPPoE works great', callback=create_wordlist_filter_from_file(f.name)) == 'Sending UDP Packets Over PPPoE Works Great'
+    f = tempfile.NamedTemporaryFile(mode='w', delete=False)  # do not delete on close
```
Comment (ppannuto) [inline titlecase/tests.py:375]:
```suggestion
    # Do not delete on close, instead do manually for Windows (see #86).
    f = tempfile.NamedTemporaryFile(mode='w', delete=False)
```

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          python setup.py install
+      - name: Tests
+        run: |
+          pip install nose coveralls
+          coverage run --source=titlecase setup.py nosetests
+
+      - name: Coveralls
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.COVERALLS_TOKEN }}
```
Comment (ppannuto) [inline .github/workflows/coverage.yml:32]:
```suggestion
          github-token: ${{ secrets.github_token }}
```

Looks like this workflow actually wants the github token, not the coveralls token?

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          python setup.py install
+      - name: Tests
+        run: |
+          pip install nose coveralls
+          coverage run --source=titlecase setup.py nosetests
+
+      - name: Coveralls
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.github_token }}
```
Comment (ppannuto) [inline .github/workflows/coverage.yml:32]:
```suggestion
          github-token: ${{ secrets.github_token }}
          path-to-lcov: ${{ github.workspace }}/coverage/lcov.info
```

File: .github/workflows/coverage.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          path-to-lcov: ${{ github.workspace }}/coverage/lcov.info
```
Comment (ppannuto) [inline .github/workflows/coverage.yml:33]:
```suggestion
      - name: Coverage
        run: |
          pip install nose coveralls
          coverage run --source=titlecase setup.py nosetests
          coveralls
        with:
          github-token: ${{ secrets.github_token }}
```

**Comment by ppannuto** [comment]
Why is this _so_ hard to configure coveralls? WIP status, got something uploading, but it seems to be an empty job: https://coveralls.io/jobs/82790913

**Comment by brocksam** [comment]
Sorry @ppannuto, I don't think I can help more specifically with the Coveralls stuff and I've not used it before. I only made the GitHub Actions workflow for it so that we could fully transfer over from Travis CI and did so just by using the Coveralls workflow docs.

**Comment by ppannuto** [comment]
No worries; maybe the answer is to abandon coveralls and move to something that more easily integrates with GH actions. I'm not attached to coveralls at all, just want _something_ that generates a coverage report. Will try to look around lunchtime for a solution hopefully.

**Pull request merged**